app/models/section_interaction.rb

```diff
@@ -79,4 +80,15 @@ def resources
   def set_state_to_review_pending
     self.state = 'review_pending'
   end
+
+  def bot_or_self_section_interaction
+    if self.track.mentor.email.split('@')[0] == 'bot' && self.todos.length.zero?
+      options = YAML.load(File.open('config/todo_data.yml'))
+      options['section_one'].each do |todo|
+        self.todos.create!(content: todo['content'])
+      end
+    elsif self.track.mentor_id == self.track.mentee_id
```

Comment:
@RushilBasappa Here instead of comparing the `mentee_id` and `mentor_id` here, move this logic to a track model method called `self_paced?`